v3/src/components/graph/hooks/use-dot-plot-responders.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect } from "react"
+import { mstReaction } from "../../../utilities/mst-reaction"
+import { comparer } from "mobx"
+import { useDataSetContext } from "../../../hooks/use-data-set-context"
+import { useGraphContentModelContext } from "./use-graph-content-model-context"
+import { useGraphDataConfigurationContext } from "./use-graph-data-configuration-context"
+
+export const useDotPlotResponders = (refreshPointPositions: (selected: boolean) => void) => {
+  const graphModel = useGraphContentModelContext()
+  const dataset = useDataSetContext()
+  const dataConfig = useGraphDataConfigurationContext()
+
+  // Respond to binAlignment and binWidth changes. We include both the volatile and non-volatile versions of these
+  // properties. Changes to the volatile versions occur during bin boundary dragging and result in the appropriate
+  // behavior during a drag. Changes to the non-volatile versions occur when a drag ends (or the user sets the bin
+  // and alignment values via the form fields) and result in the behavior required when bin boundary dragging ends.
+  useEffect(function respondToGraphBinSettings() {
+    return mstReaction(
+      () => [graphModel._binAlignment, graphModel._binWidth, graphModel.binAlignment, graphModel.binWidth],
+      () => refreshPointPositions(false),
+      {name: "respondToGraphBinSettings", equals: comparer.structural}, graphModel)
+  }, [dataset, graphModel, refreshPointPositions])
+
+  // Initialize binWidth and binAlignment on the graph model if they haven't been defined yet.
+  // This can happen when a CODAP document containing a graph with binned points is imported.
+  useEffect(function setInitialBinSettings() {
+    if (!dataConfig) return
+    if (graphModel.binWidth === undefined || graphModel.binAlignment === undefined) {
+      const { binAlignment, binWidth } = graphModel.binDetails({ initialize: true })
+      graphModel.applyModelChange(() => {
+        graphModel.setBinWidth(binWidth)
+        graphModel.setBinAlignment(binAlignment)
+      })
+    }
+  })
```

Comment:
Probably doesn't matter in this case, but was leaving off the dependencies intentional? Because it bails quickly it should be fine if it runs after every render, but you could minimize the number of times it runs with something like:
```suggestion
  }, [dataConfig, graphModel, graphModel.binAlignment, graphModel.binWidth])
```

Reply:
@kswenson Yes, I left off the dependencies because I believe this should only ever have to run once when a document is imported. There's a note about that in the comment that precedes the `useEffect`. Does that seem reasonable?